tests/testsuite/check_cfg.rs

```diff
@@ -458,3 +458,173 @@ fn features_doc() {
         .with_stderr_contains(x!("rustdoc" => "values" of "feature" with "default" "f_a" "f_b"))
         .run();
 }
+
+#[cargo_test]
+fn build_script_feedback() {
+    if !is_nightly() {
+        // rustc-check-cfg: is only availaible on nightly
+        return;
+    }
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                authors = []
+                build = "build.rs"
+            "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .file(
+            "build.rs",
+            r#"fn main() { println!("cargo:rustc-check-cfg=names(foo)"); }"#,
+        )
+        .build();
+
+    p.cargo("build -v -Zcheck-cfg=output")
+        .masquerade_as_nightly_cargo()
+        .run();
```

Comment:
Can this add a check that it is passing the flag appropriately?

I think that would be something like `.with_stderr_contains(x!("rustc" => "names" of "foo"))` with what you have here?